test/mcmc/hmc.jl

```diff
@@ -72,13 +76,15 @@ using Turing
         model_f = hmcmatrixsup()
         n_samples = 1_000
         vs = map(1:3) do _
-            chain = sample(rng, model_f, HMC(0.15, 7; adtype=adbackend), n_samples)
+            chain = sample(copy(rng), model_f, HMC(0.15, 7; adtype=adbackend), n_samples)
             r = reshape(Array(group(chain, :v)), n_samples, 2, 2)
             reshape(mean(r; dims=1), 2, 2)
         end
 
+        # TODO(mhauru) This test needs a comment explaining what is being tested.
         @test maximum(abs, mean(vs) - (7 * [1 0.5; 0.5 1])) <= 0.5
```

Comment:
The test is to make sure that the samples obey the mean of the Wishart distribution (if `x ~ Wishart(V, n)`, then `E[X] = V * n`).

I don't fully understand this testset, though. Why are we sampling 3 chains? It seems unnecessary. Especially if we're using StableRNGs, we shouldn't need to sample so much and then average

Could we replace the contents of the testset with

```julia
V = 7; n = [1 0.5; 0.5 1]
@model hmcmatrixsup() = v ~ Wishart(V, n)
model_f = hmcmatrixsup()
n_samples = 1_000

## or maybe StableRNG(23) if you agree with my other comment
chain = sample(copy(rng), model_f, HMC(0.15, 7; adtype=adbackend), n_samples)
r = reshape(Array(group(chain, :v)), n_samples, 2, 2)
r_mean = dropdims(mean(r; dims=1); dims=1)

# Check that the samples obey the mean of the Wishart distribution
@test maximum(abs.(r_mean - (V * n))) < 0.2
```

I tested locally with `StableRNG(23)` and the MAE on the mean was 0.215259. I'll leave it to you to choose an appropriate tolerance 😄

Reply:
I suppose this whole test should really go once we have a better set of DPPL models sampling from a range of distribtions. Just thinking about how much time it'll take to go through the tests to understand their purpose is, ahh, I'll need 5 Christmases to do it.

Reply:
That's much nicer, thanks. I simplified it a tiny bit more and made the change.